src/mssqlProtocolHandler.ts

```diff
@@ -0,0 +1,182 @@
+		const persistSecurityInfo = isNaN(persistSecurityInfoValue) ? undefined : persistSecurityInfoValue === 1;
+
+		const columnEncryptionSetting = args.get('columnEncryptionSetting') ?? undefined;
+		const attestationProtocol = args.get('attestationProtocol') ?? undefined;
+		const enclaveAttestationUrl = args.get('enclaveAttestationUrl') ?? undefined;
+
+		const connectTimeoutValue = parseInt(args.get('connectTimeout'));
+		const connectTimeout = isNaN(connectTimeoutValue) ? undefined : connectTimeoutValue;
+
+		const commandTimeoutValue = parseInt(args.get('commandTimeout'));
+		const commandTimeout = isNaN(commandTimeoutValue) ? undefined : commandTimeoutValue;
+
+		const connectRetryCountValue = parseInt(args.get('connectRetryCount'));
+		const connectRetryCount = isNaN(connectRetryCountValue) ? undefined : connectRetryCountValue;
+
+		const connectRetryIntervalValue = parseInt(args.get('connectRetryInterval'));
+		const connectRetryInterval = isNaN(connectRetryIntervalValue) ? undefined : connectRetryIntervalValue;
+
+		const applicationName = args.get('applicationName') ? `${args.get('applicationName')}-azdata` : 'azdata';
+
+		const workstationId = args.get('workstationId') ?? undefined;
+		const applicationIntent = args.get('applicationIntent') ?? undefined;
+		const currentLanguage = args.get('currentLanguage') ?? undefined;
+
+		const poolingValue = parseInt(args.get('pooling'));
+		const pooling = isNaN(poolingValue) ? undefined : poolingValue === 1;
+
+		const maxPoolSizeValue = parseInt(args.get('maxPoolSize'));
+		const maxPoolSize = isNaN(maxPoolSizeValue) ? undefined : maxPoolSizeValue;
+
+		const minPoolSizeValue = parseInt(args.get('minPoolSize'));
+		const minPoolSize = isNaN(minPoolSizeValue) ? undefined : minPoolSizeValue;
+
+		const loadBalanceTimeoutValue = parseInt(args.get('loadBalanceTimeout'));
+		const loadBalanceTimeout = isNaN(loadBalanceTimeoutValue) ? undefined : loadBalanceTimeoutValue;
+
+		const replicationValue = parseInt(args.get('replication'));
+		const replication = isNaN(replicationValue) ? undefined : replicationValue === 1;
+
+		const attachDbFilename = args.get('attachDbFilename') ?? undefined;
+		const failoverPartner = args.get('failoverPartner') ?? undefined;
+
+		const multiSubnetFailoverValue = parseInt(args.get('multiSubnetFailover'));
+		const multiSubnetFailover = isNaN(multiSubnetFailoverValue) ? undefined : multiSubnetFailoverValue === 1;
+
+		const multipleActiveResultSetsValue = parseInt(args.get('multipleActiveResultSets'));
+		const multipleActiveResultSets = isNaN(multipleActiveResultSetsValue) ? undefined : multipleActiveResultSetsValue === 1;
+
+		const packetSizeValue = parseInt(args.get('packetSize'));
+		const packetSize = isNaN(packetSizeValue) ? undefined : packetSizeValue;
+
+		const typeSystemVersion = args.get('typeSystemVersion') ?? undefined;
+		const connectionString = args.get('connectionString') ?? undefined;
```

Comment:
Maybe add a validation that checks if connection string is provided and other parameters should be empty?